### ISSUE TITLE
Google Cloud Storage (GCS) output plugin

### ISSUE THREAD

**Issue by epsteina16** [fluent/fluent-bit#9022]
**Is your feature request related to a problem? Please describe.**
We would like to write logs from Fluent Bit to GCS from applications running in Google's GKE. The log volume we are working is substantial and we don't require advanced searching/indexing on these logs. GCS is the ideal storage for us in this situation.

This plugin would be very similar to the S3 output plugin and the Azure Blob output plugin.
A GCS output plugin has come up many times in previous issues:
https://github.com/fluent/fluent-bit/discussions/6636
https://github.com/fluent/fluent-bit/issues/8525
https://stackoverflow.com/questions/72467592/fluentbit-to-gcs

**Describe the solution you'd like**
A new output plugin intended for uploading objects to GCS, that has similar feature parity to the Azure Blob and S3 output plugins.

**Describe alternatives you've considered**
Although GCS is S3 compatible, the default credentials chain is different. The S3 plugin fetches credentials via: https://github.com/fluent/fluent-bit-docs/blob/43c4fe134611da471e706b0edb2f9acd7cdfdbc3/administration/aws-credentials.md#aws-credentials while GCS has https://cloud.google.com/docs/authentication/application-default-credentials

**Comment by junyahondarubrik** [comment]
Hi @epsteina16 ,
Did  you find a solution for this? I also have a very similar requirement, and wanted to know.
As the original poster, do you have the permission to remove the stale label?

Thank you!

**Comment by epsteina16** [comment]
No. It would be great to hear some input from @agup006 on prioritizing this.
